examples/expressions/12_scikit_learn_estimators_in_expressions.py

```diff
@@ -1,292 +0,0 @@
-"""
-Supervised estimators and cross-validation
-------------------------------------------
-
-We can use ``.skb.apply()`` to add scikit-learn transformers to the computation,
-but also to add a supervised learner like ``HistGradientBoostingClassifier``
-or ``LogisticRegression``.
-
-When we have built a full estimator, we want to run cross-validation to
-measure its performance. We may also want to tune hyperparameters (shown in
-the next example).
-
-Skrub can run the cross-validation for us. In order to do so, it needs to
-split the dataset into training and testing sets. For this to happen, we need
-to tell skrub which items in our pipeline constitute the feature matrix ``X``
-and the targets (labels, outputs) ``y``.
-
-Indeed, an estimator built from a skrub expression can accept inputs that are
-not yet neatly organized into correctly-aligned ``X`` and ``y`` matrices.
-There may be some steps (such as loading data, performing joins or
-aggregations, separating features and targets from a single source) that need
-to be performed in order to construct the design matrix the targets.
-
-To indicate which intermediate results to split, we call ``.skb.mark_as_X()``
-and ``.skb.mark_as_y()`` on the appropriate objects:
-"""
-
-# %%
-from sklearn.ensemble import HistGradientBoostingRegressor
-
-import skrub.datasets
-
-dataset = skrub.datasets.fetch_employee_salaries()
-
-full_data = skrub.var("data", dataset.employee_salaries)
-
-employees = full_data.drop(columns="current_annual_salary").skb.mark_as_X()
-salaries = full_data["current_annual_salary"].skb.mark_as_y()
-
-vectorizer = skrub.TableVectorizer(
-    high_cardinality=skrub.MinHashEncoder(n_components=8)
-)
-predictions = employees.skb.apply(vectorizer).skb.apply(
-    HistGradientBoostingRegressor(), y=salaries
-)
-predictions
-
-# %%
-# Note how ``apply`` works for supervised estimators: we pass the targets
-# (another skrub expression) as the argument for the parameter ``y``.
-#
-# If you unfold the dropdown to see the computation graph, you will see the
-# inner nodes that have been marked as ``X`` and ``y``: these nodes are highlighted
-# in blue and red, respectively. Furthermore, the ``data`` is an input variable,
-# so it is marked with a double box.
-#
-# To perform
```

Comment:
I think there may be a bit of explanation about mark_as_x etc to rescue out of this section and move elswhere but I'm not sure

Reply:
That depends on the role we want the User Guide to play. I tried to keep it as light as possible so that readers understand the main expression concepts, without diving straight into all the details (the example showcases these details better IMO because they are built around complete use-cases). For instance, if I add `mark_as_X` in the User Guide, I should also add the `choose_from` mechanism, etc. Does that make sense?